src/main/java/com/faforever/client/game/GameService.java

```diff
@@ -168,11 +173,31 @@ public GameService(ClientProperties clientProperties, FafService fafService,
         return;
       }
 
-      JavaFxUtil.addListener(newValue.statusProperty(), new WeakChangeListener<>((observable1, oldValue1, newValue1) -> {
-        if (newValue1 == GameStatus.CLOSED) {
-          onCurrentGameEnded();
+      ChangeListener<Number> numberOfPlayersChangedListener = new ChangeListener<Number>() {
+        @Override
+        public void changed(ObservableValue<? extends Number> observable, Number oldValue, Number newValue) {
+          discordRichPresenceService.updatePlayedGameTo(currentGame.get(), playerService.getCurrentPlayer().orElseThrow(() -> new IllegalStateException("player must be set")).getId());
         }
-      }));
+      };
+
+      JavaFxUtil.addListener(newValue.numPlayersProperty(), numberOfPlayersChangedListener);
+
+      ChangeListener<GameStatus> currentGameStatusListener = new ChangeListener<GameStatus>() {
+        @Override
+        public void changed(ObservableValue<? extends GameStatus> observable1, GameStatus oldStatus, GameStatus newStatus) {
```

Comment:
I don't have my IDE but I assume that this could be a lambda instead?

Reply:
No that is not possible because the listener contains "this" and would reference the outter class in a lambda expression